File: Backend/models/noticia.py
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional , List
from datetime import datetime
from fastapi import UploadFile

class NoticiaBase(SQLModel):
    nombre: str
    descripcion: str
    imagen: str

class Noticia(NoticiaBase, table=True):
    IdNoticia: Optional[int] = Field(default= None, primary_key=True)    
    fechaPublicacion: datetime = Field(default_factory=datetime.today)

class NoticiaCreate(SQLModel):
    nombre: str
    descripcion: str
    imagen: UploadFile

class NoticiaPublic(NoticiaBase):
    IdNoticia: int
    fechaPublicacion: datetime

class NoticiaUpdate(SQLModel):
    nombre: Optional[str] = None
    descripcion: Optional[str] = None
    imagen: Optional[UploadFile] = None

File: flutter/lib/data/models/noticia.dart
class Noticia {
  final int idNoticia;
  final String nombre;
  final String descripcion;
  final String imagen;
  final DateTime fechaPublicacion;

  Noticia({
    required this.idNoticia,
    required this.nombre,
    required this.descripcion,
    required this.imagen,
    required this.fechaPublicacion,
  });

  factory Noticia.fromJson(Map<String, dynamic> json) {
    return Noticia(
      idNoticia: json['IdNoticia'],
      nombre: json['nombre'],
      descripcion: json['descripcion'],
      imagen: json['imagen'],
      fechaPublicacion: DateTime.parse(json['fechaPublicacion']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'descripcion': descripcion,
      'imagen': imagen,
    };
  }
}